apps/frontend/src/events/__tests__/Past.test.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { render, waitFor } from '@testing-library/react';
+import { RecoilRoot } from 'recoil';
+import {
+  Auth0Provider,
+  WhenReady,
+} from '@asap-hub/frontend/src/auth/test-utils';
+import {
+  createListEventResponse,
+  createEventResponse,
+} from '@asap-hub/fixtures';
+import { MemoryRouter, Route } from 'react-router-dom';
+
+import { refreshCalendarsState } from '../calendar/state';
+import { getEvents } from '../api';
+import { eventsState } from '../state';
+import { DEFAULT_PAGE_SIZE } from '../../hooks';
+import Past from '../Past';
+
+jest.mock('../api');
+
+const mockGetEvents = getEvents as jest.MockedFunction<typeof getEvents>;
+
+const renderEventsPastPage = async (currentTime = new Date()) => {
+  const result = render(
+    <RecoilRoot
+      initializeState={({ set, reset }) => {
+        set(refreshCalendarsState, Math.random());
+        reset(
+          eventsState({
+            currentPage: 0,
+            pageSize: DEFAULT_PAGE_SIZE,
+            after: new Date().toISOString(),
+          }),
+        );
+      }}
+    >
+      <React.Suspense fallback="loading">
+        <Auth0Provider user={{}}>
+          <WhenReady>
+            <MemoryRouter initialEntries={[{ pathname: '/' }]}>
+              <Route path="/">
+                <Past currentTime={currentTime} />
+              </Route>
+            </MemoryRouter>
+          </WhenReady>
+        </Auth0Provider>
+      </React.Suspense>
+    </RecoilRoot>,
+  );
+  await waitFor(() =>
+    expect(result.queryByText(/loading/i)).not.toBeInTheDocument(),
+  );
+  return result;
+};
+it('renders a list of event cards', async () => {
+  mockGetEvents.mockResolvedValue({
+    ...createListEventResponse(2),
+    items: createListEventResponse(2).items.map((item, index) => ({
+      ...item,
+      title: `Event title ${index}`,
+    })),
+  });
+
+  const { getAllByRole } = await renderEventsPastPage();
+  expect(
+    getAllByRole('heading', { level: 3 }).map((heading) => heading.textContent),
+  ).toEqual(['Event title 0', 'Event title 1']);
+  expect(
+    getAllByRole('heading', { level: 3 }).map(
+      (heading) => heading.closest('a')?.href,
+    ),
+  ).toMatchInlineSnapshot(`
```

Comment:
Could you split this out into a separate test (I want to delete all href tests when introducing the href package) and ideally use the `stringMatching` trick from below (shorter and doesn't break if changing URL structure)